lib/spree/shipment_notice.rb

```diff
@@ -10,13 +10,28 @@ def initialize(params)
     end
 
     def apply
-      locate ? update : not_found
+      return not_found unless locate
```

Comment:
I'd have to check what _the heck_ this `not_found returns to understand the code. Better will be more simple:
```ruby
find_shipment # (as `locate` is not very revealing)
if !shipment
  log_not_found # again, indicate the action with the `log` verb
  return
end
```